src/components/Subjects/Subject.js

```diff
@@ -35,34 +37,70 @@ const Subject = ({ id, name, elementaryScore, advancedScore, active,
               ref={node => { formInput.name = node }}
               defaultValue={name}
               onChange={() => onChange(id, formInput.name.value, elementaryScore, advancedScore, subjectsType)}
-              className={active ? 'subject-active subject-input-field' : 'subject-not-active subject-input-field'}
+              className={active ? 'subject-active subject-input-name' : 'subject-not-active subject-input-name'}
             />
 
             <Datalist subjectsType={subjectsType} />
           </td>
           <td>
             <input
               type='range'
-              defaultValue={elementaryScore}
+              value={elementaryScore}
               min='0'
               max='100'
-              ref={node => { formInput.elementaryScore = node }}
-              onChange={() => onChange(id, name, +formInput.elementaryScore.value, advancedScore, subjectsType)}
+              ref={node => { formInput.elementaryScoreRange = node }}
+              onChange={() => onChange(id, name, +formInput.elementaryScoreRange.value, advancedScore, subjectsType)}
+              className='subject-input-range'
             />
-
-            &nbsp; {elementaryScore ? `${elementaryScore}%` : 'Brak' }
+            <input
+              type='text'
+              value={elementaryScore || ''}
+              placeholder='Brak'
+              ref={node => { formInput.elementaryScoreText = node }}
+              onChange={() => {
+                if (!isNaN(formInput.elementaryScoreText.value)) {
```

Comment:
`formInput.elementaryScoreText.value` is used 3 times creating quite a long lines. consider extracting it to variable